File: client/ebooks_user/lib/models/access_rights/access_right.dart
import "package:ebooks_user/models/books/book.dart";
import "package:json_annotation/json_annotation.dart";

part "access_right.g.dart";

@JsonSerializable()
class AccessRight {
  int? userId;
  Book? book;
  DateTime? modifiedAt;
  bool? isFavorite;
  bool? isHidden;
  int? lastReadPage;

  AccessRight({
    this.userId,
    this.book,
    this.modifiedAt,
    this.isFavorite,
    this.isHidden,
    this.lastReadPage,
  });

  factory AccessRight.fromJson(Map<String, dynamic> json) =>
      _$AccessRightFromJson(json);
  Map<String, dynamic> toJson() => _$AccessRightToJson(this);
}

File: server/eBooks.Database/Models/AccessRight.cs
using System;
using System.Collections.Generic;

namespace eBooks.Database.Models;

public partial class AccessRight
{
    public int UserId { get; set; }

    public int BookId { get; set; }

    public DateTime ModifiedAt { get; set; }

    public bool IsFavorite { get; set; }

    public bool IsHidden { get; set; }

    public int LastReadPage { get; set; }

    public virtual Book Book { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}
